api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -364,8 +375,113 @@ public boolean checkRequireExpression(Extension candidate, AppContextModel conte
 			return false;
 		}
 	}
+
+    /**
+     * Gets the patient's <code>uuid</code> from a given <code>contextModel</code>.
+     * 
+     * @param contextModel
+     */
+    protected String getPatientUuid(AppContextModel contextModel) {
+    	Bindings bindings = new SimpleBindings();
+	    for (Map.Entry<String, Object> e : contextModel.entrySet()) {
+             bindings.put(e.getKey(), e.getValue());
+	    }
+        javascriptEngine.setBindings(bindings, ScriptContext.ENGINE_SCOPE);
+    	try {
+			String uuid = (String) javascriptEngine.eval("patient.uuid;");
+			if (StringUtils.isBlank(uuid)) {
+	    		throw new APIException("Patient uuid cannot be empty");
+			}
+			return uuid;
+		} catch (ScriptException e) {
+    		     throw new APIException("Failed to get patient uuid", e);
+		 }
+    }
```

Comment:
Looks like the context model holds direct references to the patient programs. See [here](https://github.com/openmrs/openmrs-module-coreapps/blob/c29dc4d9415ab8df87688a6298452088bde5f3d5/omod/src/main/java/org/openmrs/module/coreapps/page/controller/clinicianfacing/PatientPageController.java#L110). Wouldn't that be enough?